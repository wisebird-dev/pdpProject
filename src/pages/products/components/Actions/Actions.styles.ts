import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(({ spacing }) => ({
    button: {
        width: spacing(15),
        height: spacing(5),
    },
}));

export default useStyles;
