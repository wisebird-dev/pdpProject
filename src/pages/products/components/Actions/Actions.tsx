import React from 'react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Box from '@material-ui/core/Box';

import { Typography } from '@components';

import useStyles from './Actions.styles';

export interface ActionProps {
    handleAddToCard: () => void;
    handleBuyNow: () => void;
}

const Actions: React.FC<ActionProps> = ({
    handleAddToCard,
    handleBuyNow,
}: ActionProps) => {
    const classes = useStyles();

    return (
        <Box py={5}>
            <Grid
                container
                direction="row"
                justifyContent="space-between"
                alignItems="center"
                spacing={2}
            >
                <Grid item xs={6}>
                    <Button
                        data-testid="add-to-cart-btn"
                        variant="text"
                        style={{
                            backgroundColor: '#FFD814',
                        }}
                        className={classes.button}
                        onClick={handleAddToCard}
                    >
                        <Typography variant="body1">Add to Cart</Typography>
                    </Button>
                </Grid>
                <Grid item xs={6}>
                    <Button
                        data-testid="buy-now-btn"
                        variant="text"
                        style={{
                            backgroundColor: '#FFA41C',
                        }}
                        className={classes.button}
                        onClick={handleBuyNow}
                    >
                        <Typography variant="body1"> Buy Now</Typography>
                    </Button>
                </Grid>
            </Grid>
        </Box>
    );
};

export default Actions;
