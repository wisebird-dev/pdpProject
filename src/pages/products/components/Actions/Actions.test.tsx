import React from 'react';

import { render } from '@testUtils';

import Actions, { ActionProps } from './Actions';

describe('Product page actions suite', () => {
    const defaultProps: ActionProps = {
        handleAddToCard: jest.fn(),
        handleBuyNow: jest.fn(),
    };

    it('calls handleAddToCard when clicking the add to cart button', () => {
        const { getByTestId } = render(<Actions {...defaultProps} />);

        const addToCardBtn = getByTestId('add-to-cart-btn');

        addToCardBtn.click();

        expect(defaultProps.handleAddToCard).toHaveBeenCalled();
    });

    it('calls handleBuyNow when clicking the buy now button', () => {
        const { getByTestId } = render(<Actions {...defaultProps} />);

        const buyNowBtn = getByTestId('buy-now-btn');

        buyNowBtn.click();

        expect(defaultProps.handleBuyNow).toHaveBeenCalled();
    });
});
