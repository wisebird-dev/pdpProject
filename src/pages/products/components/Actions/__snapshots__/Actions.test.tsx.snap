// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page actions suite renders correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-2"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1"
        data-testid="add-to-cart-btn"
        style="background-color: rgb(255, 216, 20);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
            Add to Cart
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-1"
        data-testid="buy-now-btn"
        style="background-color: rgb(255, 164, 28);"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <p
            class="MuiTypography-root MuiTypography-body1"
          >
             Buy Now
          </p>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
